### PR TITLE
Added limit for length of username 

### PR DESCRIPTION
**Pull request by NDevTK**

### CODE REVIEW

**Comment by ajayyy** [comment]
HTML escaping breaks innerText :/

https://codepen.io/ajayyy/pen/QWLEREK

**Comment by OfficialNoob** [comment]
:(

**Comment by OfficialNoob** [comment]
Can now be added as just the length check.
I think the data should be escaped on the server side just not sure how it could be done

**Comment by ajayyy** [comment]
I don't see why it has to, but I do agree it doesn't feel right not to.

**Comment by ajayyy** [comment]
One thing though, I think we should add an error message on the client for this instead of returning generic error 400.

**Comment by OfficialNoob** [comment]
Yeah

File: index.js [hunk 1]
```diff
@@ -290,8 +290,12 @@ app.get('/api/viewedVideoSponsorTime', function (req, res) {
 app.post('/api/setUsername', function (req, res) {
     let userID = req.query.userID;
     let userName = req.query.username;
-
-    if (userID == undefined || userName == undefined || userID === "undefined") {
+    if(userName.length > 40) {
+        // Username to large
+        res.sendStatus(413);
+        return;
+    }
```
[the remark this answers is not in the document]
Comment (OfficialNoob) [reply index.js]: Done it

**Pull request closed without merging**